resources/server/lib/PromiseNetEvents/onNetPromise.ts

```diff
@@ -2,10 +2,18 @@ import { getSource } from '../../utils/miscUtils';
 import { mainLogger } from '../../sv_logger';
 import { CBSignature, PromiseEventResp, PromiseRequest } from './promise.types';
 import { ServerPromiseResp } from '../../../../typings/common';
+import { RateLimiter } from '../RateLimiter';
+import { GlobalRateLimiter } from '../GlobalRateLimiter';
 
 const netEventLogger = mainLogger.child({ module: 'events' });
 
-export function onNetPromise<T = any, P = any>(eventName: string, cb: CBSignature<T, P>): void {
+const globalRateLimiter = new GlobalRateLimiter(250);
+
+export function onNetPromise<T = any, P = any>(eventName: string, cb: CBSignature<T, P>, rateLimiter: RateLimiter = null): void {
```

Comment:
Perhaps a better solution would to ONLY use the global rate limiter. 

And instead of passing a rateLimiter to the onNetPromise, we would pass an options object?

``` Javascript
{
  rateLimit: 250
}
```